services/src/main/java/io/druid/cli/CliBroker.java

```diff
@@ -120,10 +130,52 @@ public void configure(Binder binder)
             MetricsModule.register(binder, CacheMonitor.class);
 
             LifecycleModule.register(binder, Server.class);
+
+            binder.bind(ForSideEffectsOnlyProvider.Child.class).toProvider(ForSideEffectsOnlyProvider.class).asEagerSingleton();
```

Comment:
Not `asEagerSingleton`, `asEagerSingleton` causes weird oddities in Guice land.  Use `Lifecycle.register(ForSideEffectsOnlyProvider.Child.class)` instead.